DuckDB.NET.Data/DuckDBStatementExecutor.cs

```diff
@@ -0,0 +1,116 @@
+            {typeof(short), BindInt16},
+            {typeof(int), BindInt32},
+            {typeof(long), BindInt64},
+            {typeof(float), BindFloat},
+            {typeof(double), BindDouble},
+            {typeof(string), BindString},
+        };
+
+    public static DuckDBQueryResult Execute(DuckDBNativeConnection connection, string query, DuckDBDbParameterCollection parameterCollection)
+    {
+        using var unmanagedQuery = query.ToUnmanagedString();
+        var queryResult = new DuckDBResult();
+        DuckDBPreparedStatement preparedStatement = null;
+        try
+        {
+            var result = NativeMethods.PreparedStatements.DuckDBPrepare(connection, unmanagedQuery, out preparedStatement);
+            if (!result.IsSuccess())
+            {
+                var errorMessage = NativeMethods.PreparedStatements.DuckDBPrepareError(preparedStatement).ToManagedString(false);
+                throw new DuckDBException(string.IsNullOrEmpty(errorMessage) ? "DuckDBQuery failed" : errorMessage, result);
+            }
+
+            BindParameters(preparedStatement, parameterCollection);
+            
+            result = NativeMethods.PreparedStatements.DuckDBExecutePrepared(preparedStatement, queryResult);
+            if (!result.IsSuccess())
+            {
+                var errorMessage = NativeMethods.Query.DuckDBResultError(queryResult).ToManagedString(false);
+                throw new DuckDBException(string.IsNullOrEmpty(errorMessage) ? "DuckDBQuery failed" : errorMessage, result);
+            }
+
+            var tmp = queryResult;
+            queryResult = null;
+            return new DuckDBQueryResult(tmp);
+        }
+        finally
+        {
+            preparedStatement?.Dispose();
+            if (queryResult != null)
+                NativeMethods.Query.DuckDBDestroyResult(queryResult);
+        }
+    }
+
+    private static void BindParameters(DuckDBPreparedStatement preparedStatement, DuckDBDbParameterCollection parameterCollection)
+    {
+        if (parameterCollection == null || parameterCollection.Count == 0) return;
+
+        var expectedParameters = NativeMethods.PreparedStatements.DuckDBParams(preparedStatement);
+        if (expectedParameters != parameterCollection.Count)
+            throw new InvalidOperationException($"Invalid number of parameters. Expected {expectedParameters}, got {parameterCollection.Count}");
+        
+        for (var i = 0; i < parameterCollection.Count; ++i)
+        {
+            var param = (DuckDBParameter)parameterCollection[i];
+            BindParameter(preparedStatement, i + 1, param);
+        }
+    }
+
+    private static void BindParameter(DuckDBPreparedStatement preparedStatement, long index, DuckDBParameter parameter)
+    {
+        if (parameter.Value == null)
+        {
+            NativeMethods.PreparedStatements.DuckDBBindNull(preparedStatement, index);
+            return;
+        }
+
+        var paramType = parameter.Value.GetType();
```

Comment:
Can we use `DbType` instead?

Reply:
We should probably set `DbType` in the `ctor` that accepts object value and use that.